api/processors/actions/student/transfer.py

```diff
@@ -23,7 +23,7 @@ def _transfer(self) -> None:
         self.student.groups.remove(self.from_group)
         self.from_group.students_former.add(self.student)
         self.to_group.students.add(self.student)
-        self.student.save()
+        self.from_group.save()
         self.to_group.save()
```

Comment:
As far as I understand, `.save()` saves the actual table in the database, while all changes in `._transfer()` affect only m2m relationships, i.e. only secondary tables are changed. If my understanding is correct, then calling `.save()` 2 times makes 2 unnecessary DB queries per each call. (That's why I keep asking this, as this is not about beauty of code, but about DB hits)

I think the easy way is to remove these lines and run the tests to see what happens :)

Reply:
Well, tests are passed in both cases and I did not even consider such cases, because in my case it is the same as optimizing two loops in one, which in 99% does not make a difference. And here as well, you might have 10-30 members of the group and even if you update them twice it will make no difference, especially with postgres as it has an advanced planner. So here I wanted it to be as much clearer as it can be and include both so that people do not think about internal implementation when reading this code.

Reply:
Btw, not sure I 100% understand why it works only with one `group.save()`, IN one case we add teacher to former and it is not related to other group,  magic :)

Reply:
> Well, tests are passed in both cases and I did not even consider that, because in my head it is the same as optimizing two loops in one, which in 99% does not make a difference. And here as well, you might have 10-30 members of the group and even if you update them twice it will make no difference, especially with postgres as it has an advanced planner. So here I wanted to be consistent and include both so that people do not think about internal implementation when reading this code.

OK, I see your point. I guess @Roxe322 would argue that people reading the code should either know Django or trust us in that we know Django 😆 

Since I'm starting to go too deep into the discussion, please just let me know if @Roxe322 's comments were addressed, and if so, I'll just approve.

Reply:
Yep, should be done, also added one more test to insure that you can not move to an none existing group (and remove from old one)